### DISCUSSION TITLE
Support images and other formats for icons

### DISCUSSION THREAD

**Discussion by ddkwork** [cogentcore/core#740] [General]
hello you, In addition to SVG, APP and Button, Chosenr and other controls support setting ICO, PNG and other icon formats? If not, is there a way to convert ICO and PNG to SVG?

**Comment by kkoreilly** [comment]
I will look into this soon.

**Reply by ddkwork** [reply]
![{92385C08-A08B-418d-BC10-3630016CB980}](https://github.com/cogentcore/core/assets/19886504/e888f43a-e69c-4936-8a5e-c7dd9a03e524)
Formats other than SVG will be supported to make it easier to set icons such as toolbars

**Reply by kkoreilly** [reply]
I will consider this.

**Reply by kkoreilly** [reply]
I made a tool for embedding image files into svg files, which you can use to accomplish using images for icons. To install it, run:

```sh
go install cogentcore.org/core/svg/cmd/svg@main
```

Example usage:

```sh
svg embed-image my-image.png
```

This will put create a file called `my-image.svg` that has the image embedded into it. Then, you can use that SVG file as an icon by adding the svg file to the icons filesystem. I will work on improving the way you extend the icons filesystem soon, but currently you can look at https://github.com/goki/kid/blob/main/buttons.go for an example. The most important lines in that example are:

```go
//go:embed icons/*.svg
var providerIcons embed.FS

func init() {
	icons.Icons = merged_fs.NewMergedFS(icons.Icons, grr.Log1(fs.Sub(providerIcons, "icons")))
}
```

**Reply by kkoreilly** [reply]
I improved the structure for extending the icons library, so you can just do this:

```go

//go:embed icons/*.svg
var myIcons embed.FS

func main() {
    icons.AddFS(grr.Log1(fs.Sub(myIcons, "icons")))
    // the rest of your code
}
```

And you can put all of your icons in an icons directory in your code using the `svg embed-image` command I showed in the last comment.

**Reply by kkoreilly** [reply]
And you can reference your icons using their string filenames without the extension. For example:

```go
gi.NewButton(b).SetIcon("my_icon_name")
```

**Reply by ddkwork** [reply]
> And you can reference your icons using their string filenames without the extension. For example:
> 
> ```go
> gi.NewButton(b).SetIcon("my_icon_name")
> ```

I Will try, thanks.

**Comment by ddkwork** [comment]
![Screenshot_20240122_200315_com.realvnc.viewer.android.jpg](https://github.com/cogentcore/core/assets/19886504/6ba30099-6261-44fc-a2be-2095b1f7e6fb)

well done, thanks!
